src/codelabs/null-safety.md

```diff
@@ -0,0 +1,357 @@
+---
+title: "Null safety"
+description: Learn about and practice writing null safe code in DartPad!
+js: [{url: 'https://dartpad.dev/inject_embed.dart.js', defer: true}]
+---
+{% assign useIframe = false -%}
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /(^|\n) *\/\/\s+ignore:[^\n]+\n/$1/g; /(\n[^\n]+) *\/\/\s+ignore:[^\n]+\n/$1\n/g"?>
+<?code-excerpt plaster="none"?>
+<style>
+{% comment %}
+TODO(chalin): move this into one of our SCSS files
+{% endcomment -%}
+iframe[src^="https://dartpad"] {
+  border: 1px solid #ccc;
+  margin-bottom: 1rem;
+  min-height: 150px;
+  resize: vertical;
+  width: 100%;
+}
+</style>
+
+This codelab teaches you about Dart’s null safe type system, which was introduced in Dart SDK 2.12. When you opt into null safety, types in your code are non-nullable by default, meaning that values can’t be null unless you say they can be.
+
+Using embedded DartPad editors, you can test your knowledge by running example code and completing exercises. To get the most out of this codelab, you should have some knowledge of [basic Dart syntax](/samples).
+
+This codelab covers the following material:
+
+* Nullable and non-nullable types.
+* How and when to use the null-aware operators: ? and !
```

Comment:
```suggestion
* How and when to use the null-aware operators: `?` and `!`
```

Reply:
Also, are they really called null-aware operators? Just checking. I was always using the term for `??` and `?.`.